backendFs2Rabbit/src/main/scala/com/itv/bucky/backend/fs2rabbit/Fs2RabbitAmqpClient.scala

```diff
@@ -0,0 +1,178 @@
+            .default(model.ExchangeName(name.value), exchangeTypeToFs2ExchangeType(exchangeType))
+            .copy(arguments = argumentsFromAnyRef(arguments))
+        ) *>
+          declare(bindings)
+      case decl.Binding(exchangeName, queueName, routingKey, arguments) =>
+        client.bindQueue(
+          model.QueueName(queueName.value),
+          model.ExchangeName(exchangeName.value),
+          model.RoutingKey(routingKey.value),
+          model.QueueBindingArgs(argumentsFromAnyRef(arguments))
+        )
+      case decl.ExchangeBinding(destinationExchangeName, sourceExchangeName, routingKey, arguments) => ???
+      case decl.Queue(name, isDurable, isExclusive, shouldAutoDelete, arguments) =>
+        client.declareQueue(DeclarationQueueConfig.default(model.QueueName(name.value)).copy(arguments = argumentsFromAnyRef(arguments)))
+    }
+
+  }
+
+  override def publisher(): Publisher[F, publish.PublishCommand] = (publishCommand: PublishCommand) => {
+    implicit val channel: model.AMQPChannel = publishChannel
+    println(publishCommand)
+    client
+      .createPublisher[PublishCommand](model.ExchangeName(publishCommand.exchange.value), model.RoutingKey(publishCommand.routingKey.value))
+      .flatMap(f => f(publishCommand))
+  }
+
+  override def registerConsumer(
+                                 queueName: bucky.QueueName,
+                                 handler: Handler[F, consume.Delivery],
+                                 exceptionalAction: consume.ConsumeAction,
+                                 prefetchCount: Int,
+                                 shutdownTimeout: FiniteDuration,
+                                 shutdownRetry: FiniteDuration
+                               ): Resource[F, Unit] =
+    client.createChannel(connection).flatMap { implicit channel =>
+      implicit val decoder: EnvelopeDecoder[F, consume.Delivery] = deliveryDecoder(queueName)
+      Resource.eval(client.createAckerConsumer[consume.Delivery](model.QueueName(queueName.value))).flatMap { case (acker, consumer) =>
+        consumer
+          .evalMap(delivery => handler(delivery.payload).map(_ -> delivery.deliveryTag))
+          .evalMap {
+            case (consume.Ack, tag)                => acker(model.AckResult.Ack(tag))
+            case (consume.DeadLetter, tag)         => acker(model.AckResult.NAck(tag))
+            case (consume.RequeueImmediately, tag) => ??? // TODO
+          }
+          .compile
+          .drain
+          .background
+          .map(_ => ())
+      }
+    }
+
+  override def isConnectionOpen: F[Boolean] = F.pure(connection.value.isOpen)
+
+}
+
+object Fs2RabbitAmqpClient {
+  def apply[F[_]](config: AmqpClientConfig)(implicit async: Async[F]): Resource[F, AmqpClient[F]] = {
+    val fs2RabbitConfig = Fs2RabbitConfig(
+      config.host, config.port, "/", 10.seconds, ssl = false, Some(config.username), Some(config.password), requeueOnNack = false, requeueOnReject = false, Some(10)
```

Comment:
I was thinking to get the `RequeueImmediate` working we could set `requeueOnReject` to true and use `reject` for `RequeueImmediate`.

Reply:
great idea, done